catalog/plugin/catalog-plugin-pointofcontactupdate/src/main/java/org/codice/ddf/catalog/plugin/metacard/PointOfContactUpdatePlugin.java

```diff
@@ -0,0 +1,119 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.metacard;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.PreAuthorizationPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+/**
+ * Update plugin that sets Point of Contact to the previous value if the new value is null.
+ */
+public class PointOfContactUpdatePlugin implements PreAuthorizationPlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PointOfContactUpdatePlugin.class);
+
+    @Override
+    public CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public UpdateRequest processPreUpdate(UpdateRequest input,
+            Map<String, Metacard> existingMetacards) throws StopProcessingException {
+        input.getUpdates()
+                .forEach(e -> {
+                    Metacard newMetacard = e.getValue();
+                    Metacard previous = getPreviousMetacardWithId(newMetacard.getId(),
+                            existingMetacards);
+
+                    if (previous != null
+                            && newMetacard.getAttribute(Metacard.POINT_OF_CONTACT) == null
+                            && isResourceMetacard(newMetacard)) {
+                        newMetacard.setAttribute(previous.getAttribute(Metacard.POINT_OF_CONTACT));
+                    }
+                });
+
+        return input;
+    }
+
+    @Override
+    public DeleteRequest processPreDelete(DeleteRequest input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public DeleteResponse processPostDelete(DeleteResponse input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public QueryRequest processPreQuery(QueryRequest input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public QueryResponse processPostQuery(QueryResponse input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public ResourceRequest processPreResource(ResourceRequest input)
+            throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public ResourceResponse processPostResource(ResourceResponse input, Metacard metacard)
+            throws StopProcessingException {
+        return input;
+    }
+
+    private boolean isResourceMetacard(Metacard metacard) {
+        return metacard.getTags()
```

Comment:
Yes it does. Since tags are new, from a legacy perspective we interpret no tags as a resource metacard.